packages/core/src/types.ts

```diff
@@ -324,3 +324,21 @@ export type RequestedEntitiesMessage = {
   entities: Array<Entry | Asset>;
   missingEntityIds?: string[];
 };
+
+//All the possible types that can be returned to a component prop
+export type BoundComponentPropertyTypes =
+  | string
+  | number
+  | boolean
+  | AssetFile
+  | Record<string, AssetFile | undefined>
+  | Document
+  | OptimizedImageAsset
+  | undefined;
+
+export type OptimizedImageAsset = {
+  url: string;
+  srcSet?: string[];
+  sizes?: string;
+  file: AssetFile;
```

Comment:
This might be a `string` though, if a specific URL was provided manually.
I would lean towards making this a generic `originalAsset` or `source` - hope they are not too ambiguous:

```suggestion
  originalAsset: {
    type: "ContentfulAssetFile" | "string" // could contain specific DAMs later
    value: AssetFile | string // should effectively be treated as `any`
  };
```